### PR TITLE
(CDAP-8167) Improves efficient in creating message tables

### PR DESCRIPTION
**Pull request by chtyim**
- Avoid checking the table existence on each call

### CODE REVIEW

**Comment by chtyim** [comment]
This one replaces the closed #7663

**Comment by chtyim** [comment]
Building http://builds.cask.co/browse/CDAP-DUT5317-2

File: cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java [hunk 1]
```diff
@@ -227,50 +291,74 @@ private void upgradeCoProcessor(TableId tableId, Class<? extends Coprocessor> co
...
-      // If table exists, then skip creating coprocessor etc
-      if (tableUtil.tableExists(admin, tableId)) {
-        return tableUtil.createHTable(hConf, tableId);
+  private HTableWithRowKeyDistributor createTable(TableId tableId, int splits,
+                                                  Class<? extends Coprocessor> coprocessor) throws IOException {
+
+    // Lookup the table descriptor from the cache first. If it is there, we assume the HBase table exists
+    // Otherwise, attempt to create it.
+    HTable hTable = null;
+    HTableDescriptor htd = tableDescriptors.get(tableId);
```
Comment (gokulavasan) [inline cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java:300]: Is the HTableDescriptor used in the logic?

**Comment by gokulavasan** [comment]
One question. LGTM 👍

File: cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java (hunk 1 above)
Comment (chtyim) [reply cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java:300]: yes. It is for getting the distributor bucket size:

https://github.com/caskdata/cdap/pull/7675/files/e6b438018362203cf5a68eabc82d60b3d95f4fcd#diff-9f9eef251f24002e8603485d1f069276R349

Comment (gokulavasan) [reply cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java:300]: I see ok 👍

**Comment by chtyim** [comment]
Build passed.

**Pull request merged**